cmd/karmada-search/app/karmada-search.go

```diff
@@ -52,3 +79,144 @@ capabilities such as global search and resource proxy in a multi-cloud environme
+}
+
+// `config` returns config for the api server given Options
+func config(o *options.Options, outOfTreeRegistryOptions ...Option) (*search.Config, error) {
+	// TODO have a "real" external address
+	if err := o.RecommendedOptions.SecureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{netutils.ParseIPSloppy("127.0.0.1")}); err != nil {
+		return nil, fmt.Errorf("error creating self-signed certificates: %v", err)
+	}
+
+	o.RecommendedOptions.Features = &genericoptions.FeatureOptions{EnableProfiling: false}
+
+	serverConfig := genericapiserver.NewRecommendedConfig(searchscheme.Codecs)
+	serverConfig.LongRunningFunc = customLongRunningRequestCheck(
+		sets.NewString("watch", "proxy"),
+		sets.NewString("attach", "exec", "proxy", "log", "portforward"))
+	serverConfig.OpenAPIConfig = genericapiserver.DefaultOpenAPIConfig(generatedopenapi.GetOpenAPIDefinitions, openapi.NewDefinitionNamer(searchscheme.Scheme))
+	serverConfig.OpenAPIConfig.Info.Title = "karmada-search"
+	if err := o.RecommendedOptions.ApplyTo(serverConfig); err != nil {
+		return nil, err
+	}
+
+	serverConfig.ClientConfig.QPS = o.KubeAPIQPS
+	serverConfig.ClientConfig.Burst = o.KubeAPIBurst
+
+	restMapper, err := apiutil.NewDynamicRESTMapper(serverConfig.ClientConfig)
+	if err != nil {
+		klog.Errorf("Failed to create REST mapper: %v", err)
+		return nil, err
+	}
+
+	karmadaClient := karmadaclientset.NewForConfigOrDie(serverConfig.ClientConfig)
+	factory := informerfactory.NewSharedInformerFactory(karmadaClient, 0)
+
+	var ctl *search.Controller
+	if !o.DisableSearch {
+		ctl, err = search.NewController(serverConfig.ClientConfig, factory, restMapper)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var proxyCtl *proxy.Controller
+	if !o.DisableProxy {
+		outOfTreeRegistry := make(runtime.Registry, 0, len(outOfTreeRegistryOptions))
+		for _, option := range outOfTreeRegistryOptions {
+			option(&outOfTreeRegistry)
+		}
+
+		proxyCtl, err = proxy.NewController(proxy.NewControllerOption{
+			RestConfig:        serverConfig.ClientConfig,
+			RestMapper:        restMapper,
+			KubeFactory:       serverConfig.SharedInformerFactory,
+			KarmadaFactory:    factory,
+			MinRequestTimeout: time.Second * time.Duration(serverConfig.Config.MinRequestTimeout),
+			OutOfTreeRegistry: outOfTreeRegistry,
+		})
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	config := &search.Config{
+		GenericConfig: serverConfig,
+		ExtraConfig: search.ExtraConfig{
+			KarmadaSharedInformerFactory: factory,
+			Controller:                   ctl,
+			ProxyController:              proxyCtl,
+		},
+	}
+	return config, nil
+}
+
+func customLongRunningRequestCheck(longRunningVerbs, longRunningSubresources sets.String) request.LongRunningRequestCheck {
+	return func(r *http.Request, requestInfo *request.RequestInfo) bool {
+		if requestInfo.APIGroup == "search.karmada.io" && requestInfo.Resource == "proxying" {
+			reqClone := r.Clone(context.TODO())
```

Comment:
this `context.TODO()` is new, like `ctrl+c` can't stop it, right?

Reply:
This is copied moved from option.go

Reply:
Changing logic is not the goal of this PR.